test/widgets/home_test.dart

```diff
@@ -0,0 +1,155 @@
+
+      // Collapsing the header updates inbox's internal state.
+      await tester.tap(find.byIcon(ZulipIcons.arrow_down).first);
+      await tester.pump();
+      check(find.byIcon(ZulipIcons.arrow_down)).findsNothing();
+      check(find.byIcon(ZulipIcons.arrow_right)).findsExactly(2);
+
+      // Switch to channels view.
+      await tester.tap(find.byIcon(ZulipIcons.hash_italic));
+      await tester.pump();
+      check(find.byIcon(ZulipIcons.arrow_down)).findsNothing();
+      check(find.byIcon(ZulipIcons.arrow_right)).findsNothing();
+
+      // The header should remain collapsed when we return to the inbox.
+      await tester.tap(find.byIcon(ZulipIcons.inbox));
+      await tester.pump();
+      check(find.byIcon(ZulipIcons.arrow_down)).findsNothing();
+      check(find.byIcon(ZulipIcons.arrow_right)).findsExactly(2);
+    });
+  });
+
+  group('menu', () {
+    final designVariables = DesignVariables.light();
+
+    final inboxMenuIconFinder = find.descendant(
+      of: find.byType(SingleChildScrollView),
+      matching: find.byIcon(ZulipIcons.inbox));
+    final channelsMenuIconFinder = find.descendant(
+      of: find.byType(SingleChildScrollView),
+      matching: find.byIcon(ZulipIcons.hash_italic));
+
+    Future<void> tapOpenMenu(WidgetTester tester) async {
+      await tester.tap(find.byIcon(ZulipIcons.menu));
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 250)); // wait for animation
+      check(find.byType(SingleChildScrollView)).findsOne();
+    }
+
+    void checkIconSelected(WidgetTester tester, Finder finder) {
+      check(tester.widget(finder)).isA<Icon>().color.isNotNull()
+        .isSameColorAs(designVariables.iconSelected);
+    }
+
+    void checkIconNotSelected(WidgetTester tester, Finder finder) {
+      check(tester.widget(finder)).isA<Icon>().color.isNotNull()
+        .isSameColorAs(designVariables.icon);
+    }
+
+    testWidgets('navigation states reflect on navigation menu buttons', (tester) async {
+      await prepare(tester);
+
+      await tapOpenMenu(tester);
+      checkIconSelected(tester, inboxMenuIconFinder);
+      checkIconNotSelected(tester, channelsMenuIconFinder);
+      await tester.tap(find.text('Cancel'));
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 250)); // wait for animation
+
+      await tester.tap(find.byIcon(ZulipIcons.hash_italic));
+      await tester.pump();
+
+      await tapOpenMenu(tester);
+      checkIconNotSelected(tester, inboxMenuIconFinder);
+      checkIconSelected(tester, channelsMenuIconFinder);
+    });
+
+    testWidgets('navigation menu buttons control navigation states', (tester) async {
+      await prepare(tester);
+
+      await tapOpenMenu(tester);
+      checkIconSelected(tester, inboxMenuIconFinder);
+      checkIconNotSelected(tester, channelsMenuIconFinder);
+      check(find.byType(InboxPageBody)).findsOne();
```

Comment:
```suggestion
      check(find.byType(InboxPageBody)).findsOne();
      check(find.byType(SubscriptionListPageBody)).findsNothing();
```

and similarly `check(find.byType(InboxPageBody)).findsNothing();` below. That helps the reader be sure that something really changed as a result of the tap — that it's not just that both finders already would find something and continue to do so (e.g. because we got something wrong in the interaction of the finders with the `Offstage` nature of the pages).